flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/CastFunctionITCase.java

```diff
@@ -1245,13 +1253,23 @@ private CastTestSpecBuilder fail(DataType dataType, Object src) {
             return fail(TestType.ERROR_SQL, dataType, src);
         }
 
+        private CastTestSpecBuilder failRuntime(
+                DataType dataType, Object src, Class<? extends Throwable> failureClass) {
+            this.testTypes.add(TestType.ERROR_RUNTIME);
+            this.columnTypes.add(dataType);
+            this.columnData.add(src);
+            this.expectedValues.add(failureClass);
+            return this;
+        }
+
         private CastTestSpecBuilder fail(TestType type, DataType dataType, Object src) {
             this.testTypes.add(type);
             this.columnTypes.add(dataType);
             this.columnData.add(src);
             return this;
         }
 
+        @SuppressWarnings("unchecked")
```

Comment:
nit: maybe move it on top of the statements with the cast to `Class<? extends Throwable>`